### DISCUSSION TITLE
How to control alignment and width of CtkOptionMenu?

### DISCUSSION THREAD

**Discussion by Rawalanche** [TomSchimansky/CustomTkinter#1878] [Q&A]
Hi,

I am having several separate issues trying to precisely position CtkOptionMenu in my application. Here's the code to begin with:
```
def __init__(self, master, image, image_name, thumbnail, remove_callback):
        super().__init__(master)
        self.rowconfigure(0, pad=8)
        self.grid_columnconfigure(0, weight=1)
        self.grid_columnconfigure(1, weight=0)
        self.image: Image.Image = image

        if image.mode.endswith("A"):
            self.source_channel_names = ["RGB", "RGBA", "Red", "Green", "Blue", "Alpha", "Luminance"]
        else:
            self.source_channel_names = ["RGB", "Red", "Green", "Blue", "Luminance"]

        self.image_name_label = customtkinter.CTkLabel(self, text=image_name, image=thumbnail, compound="left", padx=8, anchor="w")
        self.source_channel_menu = customtkinter.CTkOptionMenu(self, values=self.source_channel_names, command=self.on_source_channel_menu_clicked, width=10)
        self.target_channel_menu = CustomOptionMenu(self, values=["None"], command=self.on_target_channel_menu_clicked)
        self.duplicate_button = customtkinter.CTkButton(self, text="", image=ImageList.duplicate_icon, command=self.on_duplicate_button_clicked, width=32)
        self.remove_button = customtkinter.CTkButton(self, text="", image=ImageList.delete_icon, command=remove_callback, width=32)

        # List item layout.
        self.image_name_label.grid(row=0, column=0, padx=4, sticky="EW")
        self.source_channel_menu.grid(row=0, column=1, padx=4)
        self.target_channel_menu.grid(row=0, column=2, padx=4)
        self.duplicate_button.grid(row=0, column=3, padx=4)
        self.remove_button.grid(row=0, column=4, padx=4)
```
1. My first problem is that I can not find a way to enforce minimum width of the UI element. I can't find any argument that would force the option menu to respect the width instead of expanding based on the value. In the code above, the width of self.source_channel_menu is mere 10 units, but in the application, it expands:
![image](https://github.com/TomSchimansky/CustomTkinter/assets/38806207/adba4e90-324a-40e8-bc10-334f8abe0ed2)

2. My next problem is that when I specify wide enough width, I can not figure out how to define maximum width, so that the widget doesn't expand when the selected option string is wider than the width value. On the screenshot below, the option menus have width of 150, but user can enter option so long the widget uncontrollably expands:
![image](https://github.com/TomSchimansky/CustomTkinter/assets/38806207/f94df75c-5038-4d12-a4d0-7ed3c60f6075)

3. My last problem is that when I have widgets of various widths, for example as a side effect of my question 1. where there appears to be implicit minimum width I can not figure out how to control, I can't figure out how to customize alignment of these widgets. Let's say I would like to align them all to the left. I will set the anchor of the widget to "w' and when placing it into the grid, I will define sticky to "W" as well. 
```
self.source_channel_menu = customtkinter.CTkOptionMenu(self, values=self.source_channel_names, command=self.on_source_channel_menu_clicked, width=10, anchor="w")

self.source_channel_menu.grid(row=0, column=1, padx=4, sticky="W")
```
Neither of these will make the widgets actually align to the left thought, they are still aligned to the right:
![image](https://github.com/TomSchimansky/CustomTkinter/assets/38806207/77d58176-d32b-456d-bb67-0ce4790a098a)

So my questions are:
- How do I make option menu widgets (and also buttons for that matter) have fixed width less than their contents (displayed text for example)?
- How do I prevent the widgets expanding above the defined width when it is not desired if their content exceeds the defined width?
- How do I align/anchor the option menu within the grid cell?

Thank you in advance.

**Comment by ArsenScorpio05** [comment]
Helo there!

use: 'dynamic_resizing=False' inside the OptionMenu

self.source_channel_menu = customtkinter.CTkOptionMenu(self, values=self.source_channel_names, command=self.on_source_channel_menu_clicked, width=10, dynamic_resizing=False) -> dynamic_resizing=False was added
